src/lib/Libutil/misc_utils.c

```diff
@@ -1035,23 +1035,29 @@ validate_ext_auth_data(int auth_type, void *data, int data_len, char *ebuf, int
 
 /**
  *
- *	@brief break apart a comma delimited string into an array of strings
+ *	@brief break apart a delimited string into an array of strings
  *
- *	  @param[in] strlist - the comma delimited string
+ *	@param[in] strlist - the delimited string
+ *	@param[in] sep - the separator character
  *
  *	@return char **
  *
+ *	@note
+ *		The returned array of strings has to be freed by the caller.
  */
 char **
-break_comma_list(char *strlist)
+break_delimited_str(char *strlist, char delim)
```

Comment:
Since we are anyhow making this function more generic why can't  we allow multiple delimiters instead of  only one. If we do this this is inline with strtok_r/strtok which supports multiple delimiters.

Reply:
I am not aware of any use case where we would need to split the string on the basis of multiple tokens. If there is no use case today, I'd suggest that we don't do it. It will complicate things when we don't need it.

Reply:
I am actually thinking from the semantics perspective of the generic functions break_delimited_str() vs strtok()/strtok_r(). Because we anyhow internally making use of these functions which actually allow multiple delimiters and planning to make our new function super generic  I felt it would be better to add this. But as Arun said if we do not have any use case in PBS as of now where there is a need of breaking the string by one or more delimiters I am fine leaving the code as it is.